src/views/catalog/templatescatalog/utils/helpers.ts

```diff
@@ -46,3 +49,19 @@ export const filterTemplates = (templates: V1Template[], filters: TemplateFilter
 
       return aName?.localeCompare(bName);
     });
+
+export const setValueForRequiredParams = (template: V1Template): V1Template => {
+  const requiredParamsIndexes = findAllIndexes(
+    template?.parameters || [],
```

Comment:
i think u are running over the array of template?.parameters twice when u can do it only once? just run over draftTemplate.map why also the findallindexes needed here? just make check for parameter?.required in the map run and set the value if needed and return the item...

Reply:
Thanks, will take a look how to optimize this function.